### PR TITLE
[lldb][DataFormatter] Simplify libc++ std::map::iterator formatter

### PR DESCRIPTION
**Pull request by Michael137**
Depends on https://github.com/llvm/llvm-project/pull/97687

Similar to https://github.com/llvm/llvm-project/pull/97579, this patch simplifies the way in which we retrieve the key/value pair of a `std::map` (in this case of the `std::map::iterator`).

We do this for the same reason: not only was the old logic hard to follow, and encoded the libc++ layout in non-obvious ways, it was also fragile to alignment miscalculations (https://github.com/llvm/llvm-project/pull/97443); this would break once the new layout of std::map landed as part of https://github.com/llvm/llvm-project/issues/93069.

Instead, this patch simply casts the `__iter_pointer` to the `__node_pointer` and uses a straightforward `GetChildMemberWithName("__value_")` to get to the key/value we care about.

We can eventually re-use the core-part of the `std::map` and `std::map::iterator` formatters. But it will be an easier to change to review once both simplifications landed.

### CODE REVIEW

File: lldb/source/Plugins/Language/CPlusPlus/LibCxxMap.cpp [hunk 1]
```diff
@@ -456,3 +477,97 @@ lldb_private::formatters::LibcxxStdMapSyntheticFrontEndCreator(
...
+}
+
+bool lldb_private::formatters::LibCxxMapIteratorSyntheticFrontEnd::
+    MightHaveChildren() {
+  return true;
+}
+
+size_t lldb_private::formatters::LibCxxMapIteratorSyntheticFrontEnd::
+    GetIndexOfChildWithName(ConstString name) {
+  if (name == "first")
```
Comment (labath) [inline lldb/source/Plugins/Language/CPlusPlus/LibCxxMap.cpp:561]: Since `GetChildAtIndex` is forwarding to m_pair_sp, could this do the same as well?

**Review by labath — APPROVED** [review]

File: lldb/source/Plugins/Language/CPlusPlus/LibCxxMap.cpp (hunk 1 above)
Comment (Michael137) [reply lldb/source/Plugins/Language/CPlusPlus/LibCxxMap.cpp]: yup good idea!
